FSW/controllers/controller.py

```diff
@@ -0,0 +1,168 @@
+                                                    np.array(config["nom_rw_ang_vel"]),
+                                                    np.array(config["nadir_cam_dir"]),
+                                                    Magnetorquers,
+                                                    ReactionWheels)
+        """
+        inertia_tensor: np.ndarray,
+        G_mtb: np.ndarray,
+        b_cross_gain: float,
+        mtb_pd_gain: np.ndarray,
+        rw_pd_gain: np.ndarray,
+        rw_vel_gain: float,
+        target_angular_velocity: np.ndarray,
+        target_rw_ang_vel: np.ndarray,
+        Magnetorquers: list,"""
+        self.allocation_mat = np.zeros((Idx["NU"],3))
+       
+    def _load_gains(self):
+        gains = {}
+        if 'Gains' in self.config:
+            for key in self.config['Gains']:
+                gains[key] = float(self.config['Gains'][key])
+        return gains
+    
+    # define feedforward torque and state profile
+    def define_att_profile(self, date, pointing_mode, pointing_target, tgt_ang_vel, est_world_states):
+        ref_ctrl_states = np.zeros((7,))
+        
+        if pointing_target == "Sun":
+            # Define reference torque as zero
+            ref_torque = np.zeros((3,))
+            
+            # Define reference control states with constant estimated sun direction
+            # quaternion defining sun pointing direction
+            # +z pointing to self.state[13:16]
+            # +x cross product of +z and [0,0,1]
+            # +y completes the right handed coordinate system
+            z_body = -est_world_states[13:16] / np.linalg.norm(est_world_states[13:16])
+            x_body = np.cross(z_body, np.array([0,0,1]))
+            y_body = np.cross(z_body, x_body)
+            Rot_mat = np.vstack((x_body, y_body, z_body)).T
+            q_ref = rotmat2quat(Rot_mat)
+        elif pointing_target == "Nadir":
+            att_states = est_world_states[6:13]
+            # Default values if not sun pointing
+            ref_ctrl_states = np.zeros((19,))
+            ref_torque = np.zeros((3,))
+            
+        if  pointing_mode == "detumble":
+            # Default values if not sun pointing
+            ref_ctrl_states = np.zeros((19,))
+            ref_torque = np.zeros((3,))
+        elif pointing_mode == "spin-stabilized":
+            # Default values if not sun pointing
+            ref_ctrl_states = np.zeros((19,))
+            # reference angular velocity in pointing axis 
+            # ref_ctrl_states[:4]  = q_ref
+            # ref_ctrl_states[4:7] = tgt_ang_vel
+            ref_torque = np.zeros((3,))
+        elif pointing_mode == "3D-stabilized":
+            q_ref = rotmat2quat(Rot_mat)
+            ref_ctrl_states[:4] = q_ref
+            ref_torque = np.zeros((3,))
+        else:
+            raise ValueError(f"Unrecognized pointing mode: {pointing_mode}")
+            
+        return ref_torque, ref_ctrl_states
+    
```

Comment:
Rather than checking for the control alg ID, I think it makes sense to initialize a class attribute that points to the control alg callable. So something like
``` python
# on object initialization, point to the callable without passing args
if alg_id == "Bcross":
    self.my_control_alg  = self.lyapsunpointctr.get_dipole_moment_command
```